components/smalltalk/squeak4/Makefile

```diff
@@ -0,0 +1,67 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Package Makefile for the "Classic VM" Squeak Smalltalk V4 system
+# See http://squeak.org and http://squeakvm.org
+#
+
+# squeak4 uses cmake and a custom configure script (not GNU autoconf)
+BUILD_STYLE=		configure
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		squeak4
+COMPONENT_VERSION=	4.16.7
```

Comment:
I have another package called squeak5,  squeak5 is indeed built from "opensmalltalk", like "pharo", as you mention.   I did not yet build pharo, but squeak5 seems to be close to it.  The motivation here to use squeak4 and squeak5 is because currently it seems there is no such VM as "squeak" but it has forked in two different build systems : squeak4 uses "cmake" and squeak5 uses "autotools" (as in my understanding, squeak 3.x did).

Reply:
Fine, when you start to integrate it you should read about the mediator concept of pkg.
You'll need to add some mediator lines into the .p5m file manually.
But until we only have one squeak you can omit them.
So, with squeak5 or pharo in the pipeline it's perfectly ok to name your new package squeak4.